NETProvider/src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbParameterCollection.cs

```diff
@@ -90,6 +91,21 @@ public override object SyncRoot
 			get { return ((ICollection)_parameters).SyncRoot; }
 		}
 
+		internal bool? _collectionHasParameterWithUnicodeName;
```

Comment:
in tests I've also checked that flag is nullified correctly when parameters changed, the property doesn't allow such check. They both, field and property, are implementation details.